jplag.frontend-utils/src/main/java/de/jplag/Language.java

```diff
@@ -41,22 +41,23 @@ public interface Language {
     boolean hasErrors();
 
     /**
-     * Determines whether the parser provide column information.
+     * Number of tokens defined by the language. Some languages may have a fixed token set, others a dynamic one where the
+     * number of tokens may vary, as the token set is dynamically created for the parsed submissions.
      */
-    boolean supportsColumns();
-
-    /**
-     * Determines whether JPlag should use a fixed-width font in its reports.
-     */
-    boolean isPreformatted();
+    int numberOfTokens();
 
     /**
-     * Determines whether tokens from the scanner are indexed.
+     * Determines whether the parser provide column information. If that is the case, line and column indices are used
+     * instead of a single token index.
      */
-    boolean usesIndex();
+    default boolean supportsColumns() {
+        return true;
+    }
 
     /**
-     * Number of defined tokens in the scanner of the language.
+     * Determines whether a fixed-width font should be used to display that language.
      */
-    int numberOfTokens();
+    default boolean isPreformatted() {
+        return true;
+    }
 }
```

Comment:
This field is never used, maybe we just drop it too?

Reply:
No, the `isPreformatted`

Reply:
This should be used by the report viewer in future versions. Currently, the viewer always uses a fixed-width font.